server.go

```diff
@@ -625,8 +630,8 @@ func (srv *Server) serve(w *response) {
 }
 
 func (srv *Server) disposeBuffer(w *response) {
-	if w.udp != nil && cap(w.msg) == srv.UDPSize {
-		srv.udpPool.Put(w.msg[:srv.UDPSize])
+	if w.udpSession != nil {
+		srv.udpPool.Put(w.udpSession.Clear())
```

Comment:
If we were going to make this change, I would strip the udp pooling entirely out of `Server`. Pooling interfaces and the weird reuse patterns is very non-idiomatic Go. It would be much better for any pooling to be handled by external code in this case.

Reply:
Should be easy to move the pooling to the SessionUDP interface.